docs/understanding-your-pipeline/canonical-event/index.md

```diff
@@ -42,280 +48,403 @@ The sections below go over the standard fields found in all Snowplow events.
-| `se_value` | decimal | A value associated with the event / action e.g. the value of goods added-to-basket | No | 9.99 |
+When a user sends in a custom unstructured event, they do so as a JSON of name-value properties that conforms to a JSON schema defined for the event earlier.
+
+<Tabs groupId="warehouse" queryString>
+<TabItem value="rs/pg" label="Redshift/Postgres" default>
 
-#### Custom unstructured events
+The unstructured event is not part of the standard `atomic.events` table, instead, for users running on Redshift, it is shredded into its own table. The fields in this table will be determined by the JSON schema defined for the event in advance. Users can query just the table for that particular unstructured event, if that's all that's required for their analysis, or join that table back to the `atomic.events` table by
+
+```sql
+select 
+    ...
+from 
+    atomic.events ev
+left join 
+    atomic.my_example_unstructured_event_table sde
+    on sde.root_id = ev.event_id and sde.root_tstamp = ev.collector_tstamp
+```
 
 :::note
 
-Currently, custom events are not available in [BDP Cloud](/docs/getting-started-with-snowplow-bdp/cloud/index.md).
+You may need to take care of duplicate events as Snowplow uses an _at least once_ method of delivery.
 
 :::
 
-Custom unstructured events are a flexible tool that enable Snowplow users to define their own event types and send them into Snowplow.
+</TabItem>
+<TabItem value="bq" label="BigQuery">
 
-When a user sends in a custom unstructured event, they do so as a JSON of name-value properties that conforms to a JSON schema defined for the event earlier.
+The unstructured event is not part of the standard `atomic.events` columns, it is instead a `RECORD` type column with the name of the schema and the full version number. The fields in this table will be determined by the JSON schema defined for the event in advance, converted to snake case. Users can query fields in the self-describing event by extracting them using 
 
-The unstructured event is not part of the `atomic.events` table; instead, for users running on Redshift, it is shredded into its own table. The fields in this table will be determined by the JSON schema defined for the event in advance. Users can query just the table for that particular unstructured event, if that's all that's required for their analysis, or join that table back to the `atomic.events` table by
+```sql
+select
+...
+my_example_unstructured_event_1_0_0.my_custom_field,
+...
+from 
+    atomic.events
+```
+
+</TabItem>
+<TabItem value="sf" label="Snowflake">
+
+The unstructured event is not part of the standard `atomic.events` columns, it is instead a `OBJECT` type column with the name of the schema and the major version number. The fields in this table will be determined by the JSON schema defined for the event in advance. Users can query fields in the self-describing event by extracting them using 
```

Comment:
I don't think SF cares about the fields or their types of the `OBJECT` (someone may correct me), you can drill into any key you like so it's not quite determined by the JSON schema in the way RS/BQ are.